packages/connection-form/src/utils/validation.ts

```diff
@@ -288,15 +285,20 @@ function validateCertificateValidationWarnings(
   connectionString: ConnectionString
 ): ConnectionFormWarning[] {
   const warnings: ConnectionFormWarning[] = [];
-  if (
-    isSecure(connectionString) &&
-    (connectionString.searchParams.has('tlsInsecure') ||
-      connectionString.searchParams.has('tlsAllowInvalidHostnames') ||
-      connectionString.searchParams.has('tlsAllowInvalidCertificates'))
-  ) {
+  if (!isSecure(connectionString)) {
+    return [];
+  }
+
+  const tlsInsecure =
+    connectionString.searchParams.get('tlsInsecure') === 'true';
+  const tlsAllowInvalidHostnames =
+    connectionString.searchParams.get('tlsAllowInvalidHostnames') === 'true';
+  const tlsAllowInvalidCertificates =
+    connectionString.searchParams.get('tlsAllowInvalidCertificates') === 'true';
+
+  if (tlsInsecure || tlsAllowInvalidHostnames || tlsAllowInvalidCertificates) {
     warnings.push({
-      message:
-        'Disabling certificate validation is not recommended as it may create a security vulnerability',
+      message: `TLS/SSL certificate validation is disabled. For a more secure connection enable certificate validation if possible.`,
```

Comment:
I did like that this explicitly used the phrase “vulnerability”. Usually, that’s what it is, and “more secure connection” makes it sound like the connection still gives some security guarantees, which it effectively does not.

It’s nice to use positive language, but we should really push users towards never using these options.

Reply:
@mmarcon what's your opinion here? For this PR I can just keep the old copy and maybe do another one.

Reply:
What if we said something like:
```
TLS/SSL certificate validation is disabled. If possible, enable certificate validation to avoid security vulnerabilities.
```

Reply:
Going to merge this one and open another PR to discuss the copy for TLS

Reply:
I think that’s fine, yeah.

(Think about how browsers [do this](https://wrong.host.badssl.com/) – in both Chrome and Firefox you’ll need to go to an  `Advanced` section and are only then presented with the option to actually go to the site. And here, this potentially affects our customer’s production databases, where the risk probably tends to be higher-impact then a person visiting a bad-HTTPS site.)